pysoa/test/pytest_fixtures.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import (
+    absolute_import,
+    unicode_literals,
+)
+
+import importlib
+import os
+
+import pytest
+
+from pysoa.client import Client
+from pysoa.test.stub_service import stub_action
+
+
+@pytest.fixture(scope='session')
+def server_settings(server_class):
+    if server_class.use_django:
+        from django.conf import settings
+        settings = settings.SOA_SERVER_SETTINGS
+    else:
+        settings_module = os.environ.get('PYSOA_SETTINGS_MODULE', None)
+        assert settings_module, 'PYSOA_SETTINGS_MODULE environment variable must be set to run tests.'
+        try:
+            thing = importlib.import_module(settings_module)
+            settings = thing.SOA_SERVER_SETTINGS
+        except (ImportError, AttributeError) as e:
+            raise AssertionError('Could not access {}.SOA_SERVER_SETTINGS: {}'.format(settings_module, e))
+    return settings
+
+
+@pytest.fixture(scope='session')
+def service_client(server_class, server_settings):
+    return Client(
+        {
+            server_class.service_name: {
+                'transport': {
+                    'path': 'pysoa.common.transport.local:LocalClientTransport',
+                    'kwargs': {
+                        'server_class': server_class,
+                        'server_settings': server_settings,
+                    },
+                },
+            },
+        },
+    )
+
+
+@pytest.fixture(scope='session')
+def call_action(server_class, service_client):
+    """
+    Fixture to call a service action. Defaults to calling the localized service.
+    """
+    def do_call_action(action, body=None, service_name=None, **kwargs):
+        return service_client.call_action(service_name or server_class.service_name, action, body=body, **kwargs)
+
+    return do_call_action
```

Comment:
As Diego mentioned in the Phab diff for EB SOA, this is going to be problematic in PyTest 4. Additionally, I'm uncertain about the fixture name as it is, and definitely against the fixture name if necessary changes are made to be compatible with PyTest 4. This will require some changes.

I have a more foundational concern here, though. Why do we need a fixture for `call_action` at all? This is not really following the paradigm of PyTest fixtures. The `service_client` fixture should be sufficient, and people can just call `call_action` (or whatever) on that fixture (which is the way fixtures are meant to be used). I'm just not convinced we need this fixture at all.

Reply:
Diego was mistaken in the other diff (I replied there), this pattern is supported by Pytest 4. The reason I added this fixture is to enable the shortcut we use in our tests of dropping the service name, both for convenience and ease of migration between unittest-style tests and pytest-style. It's the pytest equivalent of the `call_action` method on `ServerTestCase`.

Reply:
My favorite part about it is enabling the addition of the `call_action_anon` fixture in ebsoa, as it makes it easy for devs to do the right thing (ie not generate users just to call actions, but to stub permissions, and to use an auth_token when calling actions anonymously)